pages/a/ms-azure-guest-post.html

```diff
@@ -0,0 +1,89 @@
+
+# Crate guest post for Azure - Setting up Azure for Crate
+
+Crate.io has been striving to create the most scalable database in our industry. We aim to make scaling your data from millions to billions of records and beyond, as simple as possible.
+
+We have clients who write and query gigabytes of data a minute, store terabytes a day and love the performance and stability that Crate offers. But the question always remained at the back of our minds, when taken to the extreme, what is Crate really capable of?
+
+Our aims were to:
+
+- See how Crate would distribute shards and partitions across such a large cluster
+- Examine the overhead added by a the cluster management
+- See what speeds the cluster could write data at
+- Another
+- And another
+
+Microsoft Azure were generously lent us the use of 8000 cores as the infrastructure for this test and engineer time to guide us through problems and questions we might have.
+
+Microsoft hoped to prove that their Azure platform was reliable enough for a project of such rapid scaling and resource intensive.
+
+So, what happened and how did we do it?
+
+We aimed to create a 1001 node Crate cluster and xxx. Setting up such a cluster on any kind of infrastructure requires planning and whilst cloud hosts simplify this process, understanding how to translate your requirements into their paradigms requires research and experimentation.
+
+Azure has several concepts that we needed to enable and configure to get our cluster started:
+
+- **Resource Group**: A 'container' that holds related resources of all types for an application.
+- **Storage Account**: A data storage solution allocated to you that can contain Blob, Table, Queue, and File data. Can be stored on SSDs or HDDs.
+- **Virtual Network**: A representation of a network in the cloud, where you can define DHCP address blocks, DNS settings, security policies, and routing.
+- **Subnet**: Further segmentation for virtual networks.
+- **Virtual Machine (VM)**: Represents a particular operating system or application stack.
+- **Network Interface (NIC)**: Represents a network interface that can be associated to one or more virtual machines (VM).
+- **Network Security Group (NSG)**: Contains a list of rules that allow or deny network traffic to your VMs in a Virtual Network. Can be associated with subnets or individual VM instances within that subnet.
+
+## Data Set
+
+70 terabytes of compressed data from the [common crawl database](http://commoncrawl.org/the-data/).
+
+## Plan A
+
+Our initial approach was to create a series of scripts to do setup the following:
+
+1. A 3 node [Consul](https://www.consul.io/) cluster to manage the cluster and handle service discovery.
+2. 3 Docker swarm master nodes to create VM instances quickly via docker machine.
+3. Gradually scale the cluster by creating docker instances running Crate and managed by the Swarm master.
+4. A 3 node monitoring and metrics cluster, we settled on [Ruxit](https://ruxit.com/).
```

Comment:
Ruxit agents were installed on each machine, they do not require a dedicated three nodes monitoring cluster